src/commands/node.mjs

```diff
@@ -385,6 +387,119 @@ export class NodeCommand extends BaseCommand {
     return true
   }
 
+  /**
+   * @param {string} namespace
+   * @param {string} nodeId
+   * @param {TaskWrapper} task
+   * @param {string} title
+   * @param {number} index
+   * @param {number} [status]
+   * @param {number} [maxAttempts]
+   * @param {number} [delay]
+   * @param {number} [timeout]
+   * @returns {Promise<string>}
+   */
+  async checkNetworkNodeActiveness (namespace, nodeId, task, title, index,
+    status = NodeStatusCodes.ACTIVE, maxAttempts = 120, delay = 1_000, timeout = 1_000) {
+    nodeId = nodeId.trim()
+    const podName = Templates.renderNetworkPodName(nodeId)
+    const podPort = 9_999
+    const localPort = 19_000 + index
+    task.title = `${title} - status ${chalk.yellow('STARTING')}, attempt ${chalk.blueBright(`0/${maxAttempts}`)}`
+
+    const srv = await this.k8.portForward(podName, localPort, podPort)
+
+    let attempt = 0
+    let success = false
+    while (attempt < maxAttempts) {
+      const controller = new AbortController()
+
+      const timeoutId = setTimeout(() => {
+        task.title = `${title} - status ${chalk.yellow('TIMEOUT')}, attempt ${chalk.blueBright(`${attempt}/${maxAttempts}`)}`
+        controller.abort()
+      }, timeout)
+
+      try {
+        const url = `http://${LOCAL_HOST}:${localPort}/metrics`
+        console.log(url)
+        const response = await fetch(url, { signal: controller.signal })
+
+        if (!response.ok) {
+          task.title = `${title} - status ${chalk.yellow('UNKNOWN')}, attempt ${chalk.blueBright(`${attempt}/${maxAttempts}`)}`
+          clearTimeout(timeoutId)
+          throw new Error()
+        }
+
+        const text = await response.text()
+        const statusLine = text
+          .split('\n')
+          .find(line => line.startsWith('platform_PlatformStatus'))
+
+        if (!statusLine) {
+          task.title = `${title} - status ${chalk.yellow('STARTING')}, attempt: ${chalk.blueBright(`${attempt}/${maxAttempts}`)}`
+          clearTimeout(timeoutId)
+          throw new Error()
+        }
+
+        const statusNumber = parseInt(statusLine.split(' ').pop())
+
+        if (statusNumber === status) {
+          task.title = `${title} - status ${chalk.green(NodeStatusEnums[status])}, attempt: ${chalk.blueBright(`${attempt}/${maxAttempts}`)}`
+          success = true
+          clearTimeout(timeoutId)
+          break
+        } else if (statusNumber === NodeStatusCodes.CATASTROPHIC_FAILURE) {
+          task.title = `${title} - status ${chalk.red('CATASTROPHIC_FAILURE')}, attempt: ${chalk.blueBright(`${attempt}/${maxAttempts}`)}`
+          break
+        } else if (statusNumber) {
+          task.title = `${title} - status ${chalk.yellow(NodeStatusEnums[statusNumber])}, attempt: ${chalk.blueBright(`${attempt}/${maxAttempts}`)}`
+        }
+        clearTimeout(timeoutId)
+      } catch {}
+
+      attempt++
+      clearTimeout(timeoutId)
+      await sleep(delay)
+    }
+
+    await this.k8.stopPortForward(srv)
```

Comment:
if anything happens above that would exit the method or throw an exception, then you would leave the port forward open

Reply:
the only thing that can prevent this from executing is the 'this.k8.portForward(podName, localPort, podPort)' which would mean the port is not open, the other code is inside a try catch